document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/service/impl/DocumentGeneratorServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package uk.gov.companieshouse.document.generator.core.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.companieshouse.document.generator.core.document.render.RenderDocumentRequestHandler;
+import uk.gov.companieshouse.document.generator.core.document.render.models.RenderDocumentRequest;
+import uk.gov.companieshouse.document.generator.core.document.render.models.RenderDocumentResponse;
+import uk.gov.companieshouse.document.generator.core.service.DocumentGeneratorService;
+import uk.gov.companieshouse.document.generator.core.service.models.DocumentRequest;
+import uk.gov.companieshouse.document.generator.core.service.models.DocumentResponse;
+import uk.gov.companieshouse.document.generator.interfaces.DocumentInfoService;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoRequest;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoResponse;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import static uk.gov.companieshouse.document.generator.core.DocumentGeneratorApplication.APPLICATION_NAME_SPACE;
+
+public class DocumentGeneratorServiceImpl implements DocumentGeneratorService {
+
+    private DocumentInfoService documentInfoService;
```

Comment:
Should we be calling this from the controller? That way we only call `generate` if there's document info available

Reply:
best to not have business logic in a controller. That's why it is in a service.